clippy_lints/src/methods/mod.rs

```diff
@@ -2386,31 +2393,34 @@ fn lint_search_is_some<'a, 'tcx>(
         let search_snippet = snippet(cx, search_args[1].span, "..");
         if search_snippet.lines().count() <= 1 {
             // suggest `any(|x| ..)` instead of `any(|&x| ..)` for `find(|&x| ..).is_some()`
+            // suggest `any(|..| *..)` instead of `any(|..| **..)` for `find(|..| **..).is_some()`
             let any_search_snippet = if_chain! {
                 if search_method == "find";
                 if let hir::ExprKind::Closure(_, _, body_id, ..) = search_args[1].node;
                 let closure_body = cx.tcx.hir().body(body_id);
                 if let Some(closure_arg) = closure_body.params.get(0);
                 if let hir::PatKind::Ref(..) = closure_arg.pat.node;
+                if let Some(name) = get_arg_name(&closure_arg.pat);
                 then {
-                    Some(search_snippet.replacen('&', "", 1))
+                    Some(search_snippet.replacen('&', "", 1).replace(&format!("*{}", name), &name.to_string()))
```

Comment:
Ah found it: You only trigger this lint, when the `PatKind` is `Ref(..)`. Maybe my suggestion with `get_arg_name()` was a little bit misleading. What I had in mind was: First removing `if let PatKind::Ref ...` from the `if_chain!`. In the then block you can do following:
```rust
if let hir::PatKind::Ref(..) = closure_arg.pat.node {
    Some(search_snippet.replacen('&', "", 1))
} else if let Some(name) = get_arg_name(&closure_arg.pat) {
    Some(search_snippet.replace(&format!("*{}", name), &name.as_str()))
} else {
    None
}
```